src/protocol/context.rs

```diff
@@ -118,10 +117,8 @@ impl<'a, N: Network, F: Field> ProtocolContext<'a, N, F> {
     /// ## Panics
     /// If you failed to upgrade to malicious protocol context
     #[allow(clippy::unused_async)] // eventually there will be await b/c of backpressure implementation
-    pub async fn malicious_multiply(self, _record_id: RecordId) {
-        // -> MaliciouslySecureMul<'a, N, F> {
-        let _accumulator = self.accumulator.as_ref().unwrap().clone();
-        // TODO: next diff!
-        // MaliciouslySecureMul::new(self, record_id, accumulator)
+    pub async fn malicious_multiply(self, record_id: RecordId) -> MaliciouslySecureMul<'a, N, F> {
+        let accumulator = self.accumulator.as_ref().unwrap().clone();
```

Comment:
This line is... a real mouthful. I have to admit I don't really follow all of these steps.

Reply:
You could try `self.accumulator.cloned().unwrap();` instead, I guess.

Reply:
It's fine. I'll leave it like this for now.